cmd/atlas/internal/cmdapi/schema.go

```diff
@@ -279,6 +305,68 @@ func CmdApplyRun(cmd *cobra.Command, _ []string) error {
 	return applyRun(cmd, c, ApplyFlags.DevURL, ApplyFlags.Paths, ApplyFlags.DryRun, ApplyFlags.AutoApprove, GlobalFlags.Vars)
 }
 
+// CmdCleanRun is the command executed when running the CLI with 'schema clean' args.
+func CmdCleanRun(cmd *cobra.Command, _ []string) error {
+	// Open a client to the database.
+	c, err := sqlclient.Open(cmd.Context(), CleanFlags.URL)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+	// Acquire a lock.
+	if l, ok := c.Driver.(schema.Locker); ok {
+		unlock, err := l.Lock(cmd.Context(), "atlas_migrate_execute", 0)
+		if err != nil {
+			return fmt.Errorf("acquiring database lock: %w", err)
+		}
+		// If unlocking fails notify the user about it.
+		defer cobra.CheckErr(unlock())
+	}
```

Comment:
I don't feel like it is needed for local development. We don't use it for `schema apply` as well.